functions/ebs-backup/main.go

```diff
@@ -28,32 +25,55 @@ var env = []string{
 }
 
 func init() {
-	log.SetHandler(logfmt.New(os.Stderr))
+	log.SetHandler(logfmt.New(os.Stdout))
 	log.SetLevel(log.InfoLevel)
 }
 
 func main() {
-	apex.HandleFunc(func(_ json.RawMessage, _ *apex.Context) (interface{}, error) {
-		c, err := config()
-		if err != nil {
-			return nil, err
-		}
+	lambda.Start(HandleRequest)
+}
 
-		e := engine.New(c)
+func HandleRequest() (r handler.Response, err error) {
+	c, err := config()
+	if err != nil {
+		return r, err
+	}
 
-		results, err := e.Run()
-		if err != nil {
-			return nil, err
-		}
+	e := engine.New(c)
 
-		for _, res := range results {
-			if res.Err != nil {
-				return nil, res.Err
-			}
+	results, err := e.Run()
+	if err != nil {
+		return r, err
+	}
+
+	errOccurred := false
+
+	for _, res := range results {
+		fields := log.Fields{
+			"name_tag":    e.Name,
+			"snapshot_id": res.CreatedSnapshot,
+			"volume_id":   res.VolumeID,
+		}
+		result := handler.Result{
+			Name:       e.Name,
+			SnapshotID: res.CreatedSnapshot,
+			VolumeID:   res.VolumeID,
 		}
+		if res.Err == nil {
+			log.WithFields(fields).Info("snapshot")
+		} else {
+			errOccurred = true
+			fields["error"] = res.Err.Error()
+			log.WithFields(fields).Error("snapshot")
```

Comment:
WithError(err)?

Reply:
I want to log the fields, but at an Error severity level.  Make sense?